### DISCUSSION TITLE
running scripts remotely

### DISCUSSION THREAD

**Discussion by cfritze64** [johnlindquist/kit#321] [General]
John:

I'm having such fun with ScriptKit. I've got some APIs that I want to access from iPhone or iPad, but Shortcuts doesn't allow or makes cumbersome some of the API authorizations and other complexities of the scripts I've written. So I've started playing with letting my iMac act as a ScriptKit server of sorts.

Instead using the server function you provide (which I can only figure out how to use on my local network), I now have iphone shortcuts that initiate remote ScriptKit output for me from my iMac no matter if I'm miles away.

- the iPhone shortcut accepts my query and writes a command + args into a "trigger" file in  a folder in my iCloud file store. I can run these shortcuts via Siri on iPhone.
- Hazel on iMac watches that folder. When a file appears, Hazel passes the file path to a ScriptKit script (and deletes the trigger file)
- that script unpacks the file contents, and executes a run() function with the command I packaged in the file
- the ultimate target script writes any returned data to the clipboard via the copy() function. The Paste app that's running on the mac syncs clipboard with all my other devices.
- Another option for the output is to have  the script create a new entry in Drafts with the content - this also syncs across devices
- finally, the script makes a call to the Astronote app which buzzes a notification on my watch that the job is done. The typical roundtrip is about 5 seconds. 

There are perhaps simpler ways to do this in the Apple ecosystem, but I like leveraging the ScriptKit functionality, and prefer the flexibility of coding the majority of routines without block editor interfaces.

**Comment by cfritze64** [comment]
…and I could ssh into my sleeping iMac via Shortcuts to deliver a CLI command to run a script, but the ssh connection doesn’t work consistently enough for me

**Comment by johnlindquist** [comment]
Wow, this all sounds amazing!

You're making me jealous. The biggest reason I want to ship 1.0 of Script Kit is so that _I_ can actually spend time playing with it instead of building it 😅

> Hazel watches...

I need to allow the `// Watch:` metadata to send the file/s that changed back to the script. 🤔
